tests/test_worker.py

```diff
@@ -594,6 +595,91 @@ def test_deleted_jobs_arent_executed(self):
         # Should not have created evidence of execution
         self.assertEqual(os.path.exists(SENTINEL_FILE), False)
 
+    def test_cancel_running_parent_job(self):
+        """Cancel a running parent job and verify that
+        dependent jobs are not started."""
+
+        def cancel_parent_job(job):
+            while job.is_queued:
+                time.sleep(1)
+
+            job.cancel()
+            return
+
+        q = Queue(
+            "low",
+        )
+        parent_job = q.enqueue(long_running_job, 5)
+
+        job = q.enqueue(say_hello, depends_on=parent_job)
+        job2 = q.enqueue(say_hello, depends_on=job)
+        status_thread = threading.Thread(target=cancel_parent_job, args=(parent_job,))
+        status_thread.start()
+
+        w = Worker([q])
+        w.work(
+            burst=True,
+        )
+        status_thread.join()
+
+        self.assertNotEqual(parent_job.result, None)
+        self.assertEqual(job.get_status(), "deferred")
```

Comment:
Mind using status enums for comparison like so? `self.assertEqual(job.get_status(), JobStatus.DEFERRED)`